addons/account/report/account_bank_reconciliation_report.py

```diff
@@ -0,0 +1,229 @@
+                    'id': line.id,
+                    'statement_id': line.statement_id.id,
+                    'type': 'bank_statement_id',
+                    'name': line.name,
+                    'footnotes': self._get_footnotes('bank_statement_id', line.id),
+                    'columns': [line.date, line.ref, abs(line.amount)],
+                    'level': 3,
+                })
+                outstanding_less_tot += abs(line.amount)
+            lines.append({
+                'id': line_id,
+                'type': 'line',
+                'name': "",
+                'footnotes': self._get_footnotes('line', line_id),
+                'columns': ['', '', "Total : " + str(outstanding_less_tot)]
+            })
+            line_id += 1
+
+        # Un-reconcilied bank statement lines
+        move_lines = self.env['account.move.line'].search([['move_id.journal_id', '=', self.env.context['journal_id'].id],
+                                                           ['move_id.statement_line_id', '=', False],
+                                                           ['user_type.type', '!=', "liquidity"],
+                                                           ['date', '<=', self.env.context['date_from']]])
+        unrec_tot = 0
+        if len(move_lines) > 0:
+            lines.append({
+                'id': line_id,
+                'type': 'line',
+                'name': "Un-Reconciled bank-statement lines",
+                'footnotes': self._get_footnotes('line', line_id),
+                'columns': [],
+                'level': 1,
+            })
+            line_id += 1
+            for line in move_lines:
+                amount = -line.debit if (line.debit > 0) else line.credit
+                lines.append({
+                    'id': line.id,
+                    'move_id': line.move_id.id,
+                    'type': 'move_line_id',
+                    'name': line.name,
+                    'footnotes': self._get_footnotes('move_line_id', line.id),
+                    'columns': [line.date, line.ref, amount],
+                    'level': 3,
+                })
+                unrec_tot += amount
+                line_id += 1
+            lines.append({
+                'id': line_id,
+                'type': 'line',
+                'name': "",
+                'footnotes': self._get_footnotes('line', line_id),
+                'columns': ['', '', "Total : " + str(unrec_tot)]
+            })
+            line_id += 1
+        
+        # Final
+        lines.append({
+            'id': line_id,
+            'type': 'line',
+            'name': "Final balance",
+            'footnotes': self._get_footnotes('line', line_id),
+            'columns': [self.env.context['date_from'], '', start_amount + outstanding_plus_tot - outstanding_less_tot + unrec_tot],
+            'level': 0,
+        })
+        
+        return lines
+
+    @api.model
+    def _get_footnotes(self, type, target_id):
```

Comment:
i find strange that you have to define this function. I guess it means that all other accounting reports have to define the same one, right?

I mean, it should be defined once for all reports